framework/framework.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package framework is a Functions Framework implementation for Go. It allows you to register
+// HTTP and event functions, then start an HTTP server serving those functions.
+package framework
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"reflect"
+	"runtime/debug"
+	"strings"
+)
+
+const (
+	functionStatusHeader = "X-Google-Status"
+	crashStatus          = "crash"
+	errorStatus          = "error"
+)
+
+// RegisterHTTPFunction registers fn as an HTTP function.
+func RegisterHTTPFunction(path string, fn func(http.ResponseWriter, *http.Request)) {
+	registerHTTPFunction(path, fn, http.DefaultServeMux)
+}
+
+// RegisterEventFunction registers fn as an event function. The function must have two arguments, a
+// context.Context and a struct type depending on the event, and return an error. If fn has the
+// wrong signature, RegisterEventFunction panics.
+func RegisterEventFunction(path string, fn interface{}) {
+	registerEventFunction(path, fn, http.DefaultServeMux)
+}
+
+// Start serves an HTTP server with registered functions.
+func Start(port string) error {
```

Comment:
Switch to `addr` (and don't do ":"+` below)?

Reply:
Existing framework uses PORT without the semicolon: https://github.com/GoogleCloudPlatform/functions-framework-nodejs/blob/master/src/index.ts#L67